core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala

```diff
@@ -62,6 +62,9 @@ trait JavaRDDLike[T, This <: JavaRDDLike[T, This]] extends Serializable {
   /** Set of partitions in this RDD. */
   def partitions: JList[Partition] = rdd.partitions.toSeq.asJava
 
+  /** Return the number of partitions in this RDD. */
```

Comment:
I think you should use Spark's own `@Since` tag, since it's used in MLLib and SQL: https://github.com/apache/spark/blob/31921e0f0bd559d042148d1ea32f865fb3068f38/core/src/main/scala/org/apache/spark/annotation/Since.scala